doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,231 @@
+
+      <path-to>/zephyr-worktree/zephyr (exported to CMake)
+
+      <path-to>/applications/hello_world
+
+  in this example, only ``<path-to>/zephyr-worktree/zephyr`` is exported to the CMake package
+  registry and therefore this Zephyr will be used.
+
+  If user wants to test the application with ``<path-to>/other-worktree/zephyr``, this can be done
+  by setting ``ZEPHYR_BASE=<path-to>/other-worktree/zephyr`` in environment, before running CMake.
+
+
+Zephyr CMake Package Version (Out-of-tree)
+******************************************
+
+When writing an out-of-tree application then it is possible to specify a Zephyr version number
+``x.y.z`` that must be used in order to build the application.
+
+As example:
+
+  .. code-block:: cmake
+
+     cmake_minimum_required(VERSION 3.13.1)
+     find_package(Zephyr VERSION 2.2.0)
+     project(hello_world)
+
+will require ``hello_world`` to be built with Zephyr 2.2.0 as minimum.
+CMake will search all exported candidates to find a Zephyr installation which matches this version
+criteria.
+
+Thus it is possible to have multiple Zephyr installations and have CMake automatically select
+between them based on the version number provided, see `CMake Package Version`_ for details.
+
+As example::
+
+      <path-to>/zephyr-2.a-worktree/zephyr (Exported to CMake)
+
+      <path-to>/zephyr-2.b-worktree/zephyr (Exported to CMake)
+
+      <path-to>/applications/hello_world
+
+
+in this case, there are two released versions of Zephyr installed at their own work trees.
+Version 2.a and 2.b.
+
+To ensure ``hello_world`` is built with minimum version ``2.a`` the following ``find_package``
+syntax may be used:
+
+  .. code-block:: cmake
+
+     cmake_minimum_required(VERSION 3.13.1)
+     find_package(Zephyr VERSION 2.a)
+     project(hello_world)
+
+
+Note that both ``2.a`` and ``2.b`` fulfill this requirement.
+
+CMake also supports the keyword ``EXACT``, to ensure an exact version is used, if that is required.
+In this case, the application CMakeLists.txt could be written as:
+
+  .. code-block:: cmake
+
+     cmake_minimum_required(VERSION 3.13.1)
+     find_package(Zephyr VERSION 2.a EXACT)
+     project(hello_world)
+
+In case no Zephyr is found which satisfies the version required, as example, the application specifies
+
+  .. code-block:: cmake
+
+     cmake_minimum_required(VERSION 3.13.1)
+     find_package(Zephyr VERSION 2.z)
+     project(hello_world)
+
+then an error similar to below will be printed::
+
+  Could not find a configuration file for package "Zephyr" that is compatible
+  with requested version "2.z".
+
+  The following configuration files were considered but not accepted:
+
+    <path-to>/zephyr-2.a-worktree/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake, version: 2.a.0
+    <path-to>/zephyr-2.b-worktree/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake, version: 2.b.0
+
+
+Multiple Zephyr Installations (In-work-tree)
```

Comment:
I do think I'm having trouble with these terms in-tree, in-work-tree, out-of-tree. Especially since git has a worktree concept also.

Reply:
> I do think I'm having trouble with these terms in-tree, in-work-tree, out-of-tree. Especially since git has a work-tree concept also.

Thanks for the comment.
And I completely agree. I struggled my-self to decide on best names.
So any suggestions for good naming is welcome.

How about:
1. in-tree / in-zephyr-tree
2. in-workspace / in-west-workspace
3. Freestanding application

Regarding the name, `in-west-workspace`, i'm not fully convinced myself, cause that name implies user must use `west`, but `find_package` will actually work the same, even if `west` was not used for cloning.
So `in-workspace` might be the best choice in 2., and it's shorter.

Reply:
Discussed offline:

```
1) Application inside Zephyr repository
2) Application inside Zephyr workspace
3) Freestanding application
```

Reply:
Did a little rearranging of the words:
```
1) Zephyr repository application
2) Zephyr workspace application
3) Zephyr freestanding application
```